### PR TITLE
accelerations for tree based federated xgboost

### PR DESCRIPTION
**Pull request by eordentlich**

### CODE REVIEW

File: nvflare/app_common/learners/xgboost_tree_fed_learner.py [hunk 1]
```diff
@@ -101,6 +101,8 @@ def initialize(self, parts: dict, fl_ctx: FLContext):
         self.set_lr()
         # load data, this is task-specific
         self.load_data(fl_ctx)
+        self.bst = None
+        self.global_model_as_dict = None
```
Comment (YuanTingHsieh) [inline nvflare/app_common/learners/xgboost_tree_fed_learner.py:105]: do we need `self.tree_method = tree_method` in the init function?

Comment (YuanTingHsieh) [reply nvflare/app_common/learners/xgboost_tree_fed_learner.py:105]: I mean the argument of the function.

Comment (eordentlich) [reply nvflare/app_common/learners/xgboost_tree_fed_learner.py:105]: Yes.  The conflict resolution got a little messed up, but missed this one in the previous cleanup.

**Comment by Can-Zhao** [comment]
Both cyclic and bagging work well and fast on my machine with cpu and gpu.

File: nvflare/apis/impl/controller.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def _check_inputs(task: Task, fl_ctx: FLContext, targets: Union[List[Client], Li
 
 
 class Controller(Responder, ControllerSpec, ABC):
-    def __init__(self, task_check_period=0.5):
+    def __init__(self, task_check_period=0.01):
```
Comment (ZiyueXu77) [inline nvflare/apis/impl/controller.py:69]: will this change impact other pipelines / shall we change it for all? - may need comments from Chester / Yan

File: nvflare/app_common/workflows/scatter_and_gather.py [hunk 3]
```diff
@@ -234,14 +234,14 @@ def control_flow(self, abort_signal: Signal, fl_ctx: FLContext) -> None:
 
                 self.fire_event(AppEventType.BEFORE_SHAREABLE_TO_LEARNABLE, fl_ctx)
                 self._global_weights = self.shareable_gen.shareable_to_learnable(aggr_result, fl_ctx)
-                fl_ctx.set_prop(AppConstants.GLOBAL_MODEL, self._global_weights, private=True, sticky=True)
+                fl_ctx.set_prop(AppConstants.GLOBAL_MODEL, self._global_weights, private=True, sticky=False)
```
Comment (ZiyueXu77) [inline nvflare/app_common/workflows/scatter_and_gather.py:237]: same here, scatter_and_gather is used much more often for other pipelines, not sure how much impact sticky has @YuanTingHsieh

File: nvflare/apis/impl/controller.py (hunk 2 above)
Comment (eordentlich) [reply nvflare/apis/impl/controller.py:69]: This was temp change before this param could be set from config files.  Will revert to original default.

File: nvflare/app_common/workflows/scatter_and_gather.py (hunk 3 above)
Comment (YuanTingHsieh) [reply nvflare/app_common/workflows/scatter_and_gather.py:237]: let merge this and I will do some testing on other examples to see if this is fine or not.

Thanks

**Comment by YuanTingHsieh** [comment]
I will fix the style issues in a later PR, thanks for the algo. update

**Pull request merged**